apps/web/components/booking/pages/BookingPage.tsx

```diff
@@ -294,6 +294,46 @@ const BookingPage = ({
   }
 
   const bookEvent = (booking: BookingFormValues) => {
+    const bookingCustomInputs = Object.keys(booking.customInputs || {}).map((inputId) => ({
+      label: eventType.customInputs.find((input) => input.id === parseInt(inputId))?.label || "",
```

Comment:
Here we're traversing  `booking.customInputs` again to find an input.

Reply:
Hopefully soon we can iradicate this behaviour when we move the BookingCustomInputs away from a json field and into an actual table.

This was discussed in https://github.com/calcom/cal.com/discussions/5840